lib/model/folder_sendrecv.go

```diff
@@ -1389,12 +1392,20 @@ func (f *sendReceiveFolder) pullerRoutine(in <-chan pullBlockState, out chan<- *
 	requestLimiter := newByteSemaphore(f.PullerMaxPendingKiB * 1024)
 	wg := sync.NewWaitGroup()
 
+	// To set the folder state to syncing when the first block is started.
+	active := false
+
 	for state := range in {
 		if state.failed() != nil {
 			out <- state.sharedPullerState
 			continue
 		}
 
+		if !active {
+			f.setState(FolderSyncing)
```

Comment:
Make setState noop if there is no transition?

Reply:
Its done once per file, makes it less ugly to read

Reply:
Once per block, but the general point still stands: readability over untested, perceived, probably negligible performance ;)

Reply:
Can we not make the folder state lockless with atomics?

Reply:
Not really, at least we wouldn't get rid of the lock, as it also protects two other `folderState` members, which must be changed atomically together. Making just `current` atomic would allow to take the shortcut return in `setState` without acquiring the lock, but that would be a bug waiting to happen, when someone will assign to `current` while holding the locking without using `atomic`.